File: androidApp/src/androidMain/kotlin/com/smjcco/wxpusher/page/web/bridge/WxpBridgeMessageParser.kt
package com.smjcco.wxpusher.page.web.bridge

import com.google.gson.JsonObject
import com.smjcco.wxpusher.base.common.WxpLogUtils
import com.smjcco.wxpusher.utils.GsonUtils

class WxpBridgeMessageParser {
    fun parse(messageJson: String): BridgeRequest? {
        return try {
            val messageBody = GsonUtils.toObj(messageJson, JsonObject::class.java) ?: return null
            val action = messageBody.get("action")?.asString ?: return null
            val callbackId = if (messageBody.has("callbackId") && !messageBody.get("callbackId").isJsonNull) {
                messageBody.get("callbackId").asString
            } else {
                null
            }
            val dataElement = messageBody.get("data")
            val dataMap = if (dataElement != null && dataElement.isJsonObject) {
                GsonUtils.jsonToObj(dataElement.asJsonObject, Map::class.java) as? Map<String, Any?>
                    ?: emptyMap()
            } else {
                emptyMap()
            }

            BridgeRequest(
                action = action,
                data = dataMap,
                callbackId = callbackId
            )
        } catch (e: Exception) {
            WxpLogUtils.w(message = "处理WebBridge消息失败", throwable = e)
            null
        }
    }
}

File: iosApp/wxpusher/Page/WebView/bridge/WxpBridgeMessageParser.swift
import Foundation

final class WxpBridgeMessageParser {
    func parse(_ body: Any) -> WxpBridgeRequest? {
        if let mapBody = body as? [String: Any] {
            return parseDictionary(mapBody)
        }
        if let jsonBody = body as? String,
           let data = jsonBody.data(using: .utf8),
           let jsonObj = try? JSONSerialization.jsonObject(with: data) as? [String: Any] {
            return parseDictionary(jsonObj)
        }
        return nil
    }

    private func parseDictionary(_ body: [String: Any]) -> WxpBridgeRequest? {
        guard let action = body["action"] as? String else {
            return nil
        }
        return WxpBridgeRequest(
            action: action,
            data: body["data"] as? [String: Any] ?? [:],
            callbackId: body["callbackId"] as? String
        )
    }
}
